src/environmentd/src/http/sql.rs

```diff
@@ -68,8 +69,15 @@ pub async fn handle_sql_ws(
 #[derive(Serialize, Deserialize, Debug)]
 #[serde(untagged)]
 pub enum WebSocketAuth {
-    Basic { user: String, password: String },
-    Bearer { token: String },
+    Basic {
+        user: String,
+        password: String,
+        options: Option<BTreeMap<String, String>>,
```

Comment:
(asked elsewhere): Does this need to be an Option, or is there a way for serde to have not-present mean empty? Is that a good idea?

Reply:
There is! Updated such that if `options` is not present, we'll deserialize to an empty `BTreeMap`. IMO this is a good idea because when using the map we were already doing `options.unwrap_or_default()`, thus essentially treating no options and empty options as the same. In the future if we care about no options vs empty options we can change this